### PR TITLE
feat: condKL_compProd_meas

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
Add the lemma condKL_compProd_meas, about the kl divergence conditional to a compProd.
This PR depends on #16 and #17.

### CODE REVIEW

**Comment by RemyDegenne** [comment]
Merge master?

**Comment by LorenzoLuccioli** [comment]
We can merge, but I don't have permission to do it

**Comment by RemyDegenne** [comment]
Sorry my comment was not clear: I was asking you to merge master into that branch, now that the dependencies are merged, in order to reduce the diff and check for conflicts.

File: TestingLowerBounds/KullbackLeibler.lean [hunk 1]
```diff
@@ -387,6 +387,101 @@ lemma condKL_const {ξ : Measure β} [IsFiniteMeasure ξ] [IsFiniteMeasure μ] [
...
+    constructor
+    · filter_upwards [h_ae, h_int1, h_int2.1] with a ha_ae ha_int ha_int2
+      apply condKL_ne_top_iff.mpr
+      simp only [kernel.snd'_apply]
+      exact ⟨ha_ae, ⟨ha_int, ha_int2⟩⟩
+    · refine Integrable.congr ?_ (h_ae_eq h_ae h_int1).symm
+      replace h_int := h_int2.2
+      apply Integrable.mono h_int h_meas
+      refine ae_of_all μ ?_
+      intro a
```
Comment (RemyDegenne) [inline TestingLowerBounds/KullbackLeibler.lean]:
```suggestion
      refine ae_of_all μ fun a ↦ ?_
```

**Comment by RemyDegenne** [comment]
Is it possible to use #35 to get a much shorter proof here?

**Comment by LorenzoLuccioli** [comment]
> Is it possible to use #35 to get a much shorter proof here?

Not with the same level of generality, because in the fDiv case we require `μ` to be a finite measure, while in kl we only require it to be S finite.

File: TestingLowerBounds/KullbackLeibler.lean [hunk 2]
```diff
@@ -387,6 +387,100 @@ lemma condKL_const {ξ : Measure β} [IsFiniteMeasure ξ] [IsFiniteMeasure μ] [
...
+    condKL κ η (μ ⊗ₘ ξ) = ∫ x, (condKL (kernel.snd' κ x) (kernel.snd' η x) (ξ x)).toReal ∂μ := by
+  rw [condKL_ne_top_iff'.mp h, Measure.integral_compProd (condKL_ne_top_iff.mp h).2.2]
+  replace h := condKL_compProd_meas_eq_top.mpr.mt h
+  push_neg at h
+  norm_cast
+  apply integral_congr_ae
+  filter_upwards [h.1] with a ha
+  simp_rw [condKL_ne_top_iff'.mp ha, EReal.toReal_coe, kernel.snd'_apply]
+
+
```
Comment (RemyDegenne) [inline TestingLowerBounds/KullbackLeibler.lean]:
```suggestion
```

File: TestingLowerBounds/KullbackLeibler.lean [hunk 3]
```diff
@@ -387,6 +387,100 @@ lemma condKL_const {ξ : Measure β} [IsFiniteMeasure ξ] [IsFiniteMeasure μ] [
...
+      calc ‖∫ b, (kl (κ (a, b)) (η (a, b))).toReal ∂ξ a‖
+      _ ≤ ∫ b, ‖(kl (κ (a, b)) (η (a, b))).toReal‖ ∂ξ a :=
+        norm_integral_le_integral_norm _
+      _ = _ := by
+        simp only [norm_eq_abs]
+        apply (abs_of_nonneg _).symm
+        positivity
+
+-- TODO: find a better name
+-- TODO: generalize this to fdiv
```
Comment (RemyDegenne) [inline TestingLowerBounds/KullbackLeibler.lean]:
```suggestion
```
This is done in the other PR.

**Review by RemyDegenne — APPROVED** [review]

**Pull request merged**